plugins/csp-demo-node-editor/src/nodes/MathNode.cpp

```diff
@@ -0,0 +1,98 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// SPDX-FileCopyrightText: German Aerospace Center (DLR) <cosmoscout@dlr.de>
+// SPDX-License-Identifier: MIT
+
+#include "MathNode.hpp"
+
+#include "../logger.hpp"
+
+#include "../../../../src/cs-utils/filesystem.hpp"
+#include "../../../../src/cs-utils/utils.hpp"
+
+namespace csp::demonodeeditor {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const std::string MathNode::sName = "Math";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::string MathNode::sSource() {
+  return cs::utils::filesystem::loadToString(
+      "../share/resources/nodes/csp-demo-node-editor/MathNode.js");
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::unique_ptr<MathNode> MathNode::sCreate() {
+  return std::make_unique<MathNode>();
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::string const& MathNode::getName() const {
+  return sName;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void MathNode::process() {
+
+  // Whenever this method is called, we compute the output value based on the given input values and
+  // the currently selected math operation.
+  double first  = readInput<double>("first", 0.0);
+  double second = readInput<double>("second", 0.0);
+
+  double result = 0.0;
+
+  switch (mOperation) {
+  case Operation::eAdd:
+    result = first + second;
+    break;
+  case Operation::eSubtract:
+    result = first - second;
+    break;
+  case Operation::eMultiply:
+    result = first * second;
+    break;
+  case Operation::eDivide:
+    result = first / second;
+    break;
+  default:
+    break;
+  }
```

Comment:
This gives me flashbacks to first semester programming exercises 😋

Reply:
:rofl: